services/QuillLMS/client/app/bundles/Teacher/components/general_components/tooltip/__tests__/activity_details.test.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import moment from 'moment';
+import { render, screen } from "@testing-library/react";
+
+import ActivityDetails from '../activity_details';
+
+jest.mock('moment', () => ({
+  default: jest.requireActual('moment')
+}))
+
+describe('ActivityDetails component', () => {
+  const baseData: any = { publishDate: '2016-10-17 00:05:50.361093' };
+
+  it('should render', () => {
+    const { asFragment } = render(<ActivityDetails data={baseData} />);
+    expect(asFragment()).toMatchSnapshot();
+  })
+  it('should show Objectives section if activity_description present', () => {
+    baseData.activity_description = "This is a test description"
+    const { asFragment } = render(<ActivityDetails data={baseData} />);
+    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText(/objectives/i)).toBeInTheDocument()
+    expect(screen.getByText(/this is a test description/i)).toBeInTheDocument()
+  })
+  it('should show Scheduled For if scheduled is true', () => {
+    baseData.scheduled = true
+    const { asFragment } = render(<ActivityDetails data={baseData} />);
+    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText(/scheduled for/i)).toBeInTheDocument()
+    expect(screen.getByText(/this is a test description/i)).toBeInTheDocument()
+  })
+  it('should show Published if scheduled is false', () => {
+    baseData.scheduled = false
+    const { asFragment } = render(<ActivityDetails data={baseData} />);
+    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText(/published/i)).toBeInTheDocument()
+  })
+  it('should show Due if dueDate present', () => {
+    baseData.dueDate = true
+    const { asFragment } = render(<ActivityDetails data={baseData} />);
+    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText(/due/i)).toBeInTheDocument()
+  })
```

Comment:
It looks to me like it will show "Due" even if `dueDate` is `false` because we display "N/A" when it's not set.  Do we want a more specific expectation that captures something that'll change when the value is set?

Reply:
yeah, good catch-- I'll tweak this test